web/app/entry.worker.ts

```diff
@@ -0,0 +1,175 @@
+/// <reference lib="WebWorker" />
+
+// Gently provided by https://github.com/jacob-ebey/remix-pwa
+
+import { json } from '@remix-run/server-runtime';
+
+export type {};
+declare const self: ServiceWorkerGlobalScope;
+
+function debug(...messages: any[]) {
+  if (process.env.NODE_ENV === 'development') {
+    console.debug(...messages);
+  }
+}
+
+let STATIC_ASSETS = ['/build/', '/icons/'];
+
+const ASSET_CACHE = 'asset-cache';
+const DATA_CACHE = 'data-cache';
+const DOCUMENT_CACHE = 'document-cache';
+
+async function handleInstall(event: ExtendableEvent) {
+  debug('Service worker installed');
+}
+
+async function handleActivate(event: ExtendableEvent) {
+  debug('Service worker activated');
+}
+
+async function handleMessage(event: ExtendableMessageEvent) {
+  let cachePromises: Map<string, Promise<void>> = new Map();
+
+  if (event.data.type === 'REMIX_NAVIGATION') {
+    let { isMount, location, matches, manifest } = event.data;
+    let documentUrl = location.pathname + location.search + location.hash;
+
+    let [dataCache, documentCache, existingDocument] = await Promise.all([
+      caches.open(DATA_CACHE),
+      caches.open(DOCUMENT_CACHE),
+      caches.match(documentUrl),
+    ]);
+
+    if (!existingDocument || !isMount) {
+      debug('Caching document for', documentUrl);
+      cachePromises.set(
+        documentUrl,
+        documentCache.add(documentUrl).catch(error => {
+          debug(`Failed to cache document for ${documentUrl}:`, error);
+        })
+      );
+    }
+
+    if (isMount) {
+      for (let match of matches) {
+        if (manifest.routes[match.id].hasLoader) {
+          let params = new URLSearchParams(location.search);
+          params.set('_data', match.id);
+          let search = params.toString();
+          search = search ? `?${search}` : '';
+          let url = location.pathname + search + location.hash;
```

Comment:
what is the reason to use let instead of const?

Reply:
Copy and paste! 🙈